docker-java-core/src/main/java/com/github/dockerjava/core/DockerClientConfig.java

```diff
@@ -53,7 +66,74 @@ default ObjectMapper getObjectMapper() {
             .disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)
             .disable(SerializationFeature.FAIL_ON_EMPTY_BEANS);
 
+    {
+        ObjectMapper originalObjectMapper = objectMapper.copy();
+        objectMapper.registerModule(new SimpleModule("docker-java") {
+            @Override
+            public void setupModule(SetupContext context) {
+                super.setupModule(context);
+                context.addBeanDeserializerModifier(new BeanDeserializerModifier() {
+                    @Override
+                    public JsonDeserializer<?> modifyDeserializer(
+                        DeserializationConfig config,
+                        BeanDescription beanDescription,
+                        JsonDeserializer<?> originalDeserializer
+                    ) {
+                        if (!beanDescription.getType().isTypeOrSubTypeOf(DockerObject.class)) {
+                            return originalDeserializer;
+                        }
+
+                        return new DockerObjectDeserializer(
+                            originalDeserializer,
+                            beanDescription,
+                            originalObjectMapper
+                        );
+                    }
+                });
+            }
+        });
+    }
+
     public ObjectMapper getObjectMapper() {
         return objectMapper;
     }
 }
+
+class DockerObjectDeserializer extends DelegatingDeserializer {
+
+    private final BeanDescription beanDescription;
+
+    private final ObjectMapper originalMapper;
+
+    DockerObjectDeserializer(
+        JsonDeserializer<?> delegate,
+        BeanDescription beanDescription,
+        ObjectMapper originalMapper
+    ) {
+        super(delegate);
+        this.beanDescription = beanDescription;
+        this.originalMapper = originalMapper;
+    }
+
+    @Override
+    protected JsonDeserializer<?> newDelegatingInstance(JsonDeserializer<?> newDelegatee) {
+        return new DockerObjectDeserializer(newDelegatee, beanDescription, originalMapper);
+    }
+
+    @Override
+    @SuppressWarnings({"deprecation", "unchecked"})
+    public Object deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
+        JsonNode jsonNode = p.readValueAsTree();
+
+        Object deserializedObject = originalMapper.treeToValue(jsonNode, beanDescription.getBeanClass());
```

Comment:
Jackson 2.13 will  help here with method `DeserializationContext.readTreeAsValue(JsonNode, Class)`, but alas not yet available. But there is `DeserializationContext.readTree(JsonParser)` which has some small benefits over going through `JsonParser` -- probably not much difference here just minor improvement.

Reply:
@cowtowncoder thanks for having a look!
I tried `DeserializationContext.readTree(JsonParser)` but ended up getting a recursive problem 😅 Hence the "originalMapper" workaround.

Reply:
Ah. Odd... would be interested in details if you have any (just to see if there's something to fix), but makes sense to skip in that case.

Reply:
As now I have your aura around, let me try it again and report back :)
Can also share how to try it, without even having Docker running (a.k.a. in memory)

Reply:
Sounds good!

Reply:
Ah, okay, I am stupid :D I confused `readTree` with `readValue` 😅 

`readValue` gives a recursive problem, while `JsonNode jsonNode = ctxt.readTree(p);` (previous line) works just fine, thanks!